mono/metadata/metadata.h

```diff
@@ -512,6 +512,8 @@ MONO_API uint32_t      mono_metadata_decode_table_row_col (MonoImage *image, int
 					   int            idx,
 					   unsigned int          col);
 
+MONO_API void  mono_metadata_get_class_guid (MonoClass* klass, uint8_t* guid, MonoError *error);
```

Comment:
The `metadata.h` header is part of Mono's public API.  We shouldn't add stuff to it without good reason.

1. Move this declaration to `metadata-internals.h` (or `cominterop.h`)
2. Remove the `MONO_API` attribute.